www/docs/guides/walletAccount.md

```diff
@@ -0,0 +1,185 @@
+---
+sidebar_position: 9
+---
+
+# WalletAccount
+
+**Use wallets (like Braavos & ArgentX) to sign your transactions in your DAPP.**
+
+The `WalletAccount` class is similar to the regular `Account` class, but is also able to ask to a browser wallet to sign and send a transaction. Some other cool functionalities will be detailed hereunder.
+
+The private key of a WalletAccount is held in a browser wallet (as ArgentX or Braavos), and any signature is managed by the wallet. You don't have to manage in your DAPP the security of any private key.
+
+:::caution
+This class is working only in the scope of a DAPP. You can't use it in a node.js script.
+:::
+
+## Architecture
+
+![](./pictures/WalletAccountArchitecture.png)
+
+If you want to read Starknet, the WalletAccount will read directly the blockchain. That's why at the initialization of a WalletAccount, you need to put in the parameters a Provider instance. It will be used for all reading activities.
```

Comment:
```suggestion
If you want to read from Starknet, the WalletAccount should be used directly to access the blockchain. Hence, during the initialization of a WalletAccount, you need to include a Provider instance as a parameter. This will be used for all reading activities.
```

Reply:
The meaning is changed, in a wrong way. I want to say that in case of Starknet reading , WalletAccount will internally read  Starknet without any SWO use, using the RPC Provider defined at instantiation.